salt/addons/init.sls

```diff
@@ -38,6 +38,8 @@ apply-dns:
     - require:
       - kube-apiserver
       - file:      /etc/kubernetes/addons/kubedns.yaml
+    - check_cmd:
+      - kubectl get deploy kube-dns -n kube-system | grep kube-dns | cat
```

Comment:
This will cause issues during an upgrade - if the kube-dns deployment is changed (say, to use a new version of the container), this will end up not applying the new kubedns.yaml file.

Reply:
@robdaemon My understanding is that the kubectl apply will run irrespective of check_cmd's output. But, what this salt state will return is the return code from check_cmd rather than from kubectl apply. Let me know if you think otherwise.

https://docs.saltstack.com/en/latest/ref/states/requisites.html#check-cmd